### PR TITLE
Xsd date and time support

### PR DESCRIPTION
**Pull request by moihn**
I have added support to xsd:date, xsd:datetime, and xsd:time by referencing the code at  https://github.com/NYTimes/encoding-wrapper/blob/master/elementalconductor/time.go 

But it actually isn't quite fully covering the various string literals of these types as described at:
xsd:time => http://books.xmlschemata.org/relaxng/ch19-77311.html
xsd:date => http://books.xmlschemata.org/relaxng/ch19-77041.html
xsd:datetime => http://books.xmlschemata.org/relaxng/ch19-77049.html

So I used similar idea to develop a better implementation of XsdDate, XsdDateTime and XsdTime.

Please let me know if you have any comments.

The code has been tested against our product and SOAP service, and "go test" can have a PASS.

Thanks and regards

### CODE REVIEW

**Comment by c4milo** [comment]
Awesome, I will take a look at this PR during the weekend. Thank you!

File: fixtures/epcis/epcisquery.src [hunk 1]
```diff
@@ -33,7 +33,7 @@ type Document struct {
 	// The date the message was created. Used for auditing and logging.
 	//
 
-	CreationDate time.Time `xml:"creationDate,attr,omitempty" json:"creationDate,omitempty"`
+	CreationDate soap.XsdDateTime `xml:"creationDate,attr,omitempty" json:"creationDate,omitempty"`
```
Comment (c4milo) [inline fixtures/epcis/epcisquery.src:36]: In Go, the naming convention will be **XSD**DateTime

**Comment by c4milo** [comment]
LGTM! only a minor nitpick!

File: fixtures/epcis/epcisquery.src (hunk 1 above)
Comment (moihn) [reply fixtures/epcis/epcisquery.src:36]: Please find the PR for fixing it here: https://github.com/hooklift/gowsdl/pull/196

**Pull request merged**